pkg/registry/core/namespace/storage/storage_test.go

```diff
@@ -152,7 +152,7 @@ func TestDeleteNamespaceWithIncompleteFinalizers(t *testing.T) {
 	defer server.Terminate(t)
 	defer storage.store.DestroyFunc()
 	key := "namespaces/foo"
-	ctx := genericapirequest.NewContext()
+	ctx := genericapirequest.WithNamespace(genericapirequest.NewContext(), metav1.NamespaceNone)
```

Comment:
namespace is set in the context based on whether there's a `namespaces/$namespace` in the request URL. `Namespace` delete requests set the namespace name in the context.

Reply:
ah, gotcha, thanks 👍🏼 
So, in tests such as this, we use a separate context for the `Create` request and a different context for update/delete?

Reply:
to mirror what happens in real life, yeah

Reply:
pulled this commit into https://github.com/kubernetes/kubernetes/pull/94637 and addressed this in a second commit